medusa/common.py

```diff
@@ -559,7 +559,7 @@ def should_replace(ep_status, old_quality, new_quality, allowed_qualities, prefe
         :param search_type: The search type, that started this method
         :return: True if the old quality should be replaced with new quality
         """
-        if ep_status and ep_status not in (DOWNLOADED, SNATCHED, SNATCHED_PROPER):
+        if not ep_status or ep_status not in (DOWNLOADED, SNATCHED, SNATCHED_PROPER):
```

Comment:
The first condition is not needed, but maybe it is better to keep this more explicit for now (also avoids the iteration).